src/lib/core/plugin.js

```diff
@@ -160,8 +160,12 @@ Plugin.prototype.addContractFile = function(file) {
   this.addPluginType('contractFiles');
 };
 
-Plugin.prototype.registerConsoleCommand = function(cb) {
-  this.console.push(cb);
+Plugin.prototype.registerConsoleCommand = function(optionsOrCb) {
+  if (typeof optionsOrCb === 'function') {
+    this.logger.warn(__('Registering console commands with a function is deprecated'));
+    // TODO add docs on how to register
+  }
+  this.console.push(optionsOrCb);
```

Comment:
We could. However, it can be a function too and we can't know exactly from the function what it actually matches (eg: the command `resolve embark.eth` has a `matches()` function that only matches the first word)
So, it wouldn't check for duplicates really well.

Also, right now, only the last registered command is used (I think), so like user plugins, they override Embark commands, which makes sense IMO.

What do you think?

Reply:
Hmm I guess it's a matter of opinion. I'd assume certain commands should not be overridable as they are core pieces of functionality. 

As far as functions are concerned, we could check if matches is an array, and then check if any of the entries already exist.